### ISSUE TITLE
JS does not ignores transient properties in specific conditions

### ISSUE THREAD

**Issue by Dominaezzz** [Kotlin/kotlinx.serialization#1203]
**Describe the bug**
The below example produces this error.
```
TypeError: Cannot set property message of Error which has only a getter
```

**To Reproduce**
```
@Serializable
sealed class Broken : Exception() {
    override val message: String? get() = simple
    abstract val simple: String?

    @Serializable
    @SerialName("single")
    data class BrokenSubClass(override val simple: String?) : Broken()
}

@Test
fun testReproduce() {
    val json = """{ "type": "single", "simple": "Doesn't mater" }"""
    val broken = Json.decodeFromString(Broken.serializer(), json)
    println(broken)
}
```

**Expected behavior**

**Environment**
 - Kotlin version: 1.4.10
 - Library version: 1.0.1
 - Kotlin platforms: Only JS (JVM and Native are fine)
 - Gradle version: 6.4

**Comment by shanshin** [comment]
Unfortunately, this is a compiler bug: https://youtrack.jetbrains.com/issue/KT-43490.

As a workaround, you can override the `message` property in `BrokenSubClass` like this
```
        @Serializable
        @SerialName("single")
        data class BrokenSubClass(override val simple: String?) : Broken() {
            override val message: String? = super.message
        }
```

**Comment by sandwwraith** [comment]
[KT-43490](https://youtrack.jetbrains.com/issue/KT-43490) should be fixed in 1.9.0 and wasn't a serialization plugin issue anyway
